lib/gui/app/components/target-selector/target-selector-modal.tsx

```diff
@@ -0,0 +1,480 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+	faChevronDown,
+	faExclamationTriangle,
+} from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { scanner, sourceDestination } from 'etcher-sdk';
+import * as React from 'react';
+import {
+	Badge,
+	Table,
+	Txt,
+	Flex,
+	Link,
+	TableColumn,
+	ModalProps,
+} from 'rendition';
+import styled from 'styled-components';
+
+import {
+	getDriveImageCompatibilityStatuses,
+	hasListDriveImageCompatibilityStatus,
+	isDriveValid,
+	TargetStatus,
+	Image,
+} from '../../../../shared/drive-constraints';
+import { compatibility } from '../../../../shared/messages';
+import { bytesToClosestUnit } from '../../../../shared/units';
+import { getDrives, hasAvailableDrives } from '../../models/available-drives';
+import {
+	getImage,
+	getSelectedDrives,
+	isDriveSelected,
+} from '../../models/selection-state';
+import { store } from '../../models/store';
+import { logEvent, logException } from '../../modules/analytics';
+import { open as openExternal } from '../../os/open-external/services/open-external';
+import { Modal } from '../../styled-components';
+
+interface UsbbootDrive extends sourceDestination.UsbbootDrive {
+	progress: number;
+}
+
+interface DriverlessDrive {
+	displayName: string; // added in app.ts
+	description: string;
+	link: string;
+	linkTitle: string;
+	linkMessage: string;
+	linkCTA: string;
+}
+
+type Target = scanner.adapters.DrivelistDrive | DriverlessDrive | UsbbootDrive;
+
+function isUsbbootDrive(drive: Target): drive is UsbbootDrive {
+	return (drive as UsbbootDrive).progress !== undefined;
+}
+
+function isDriverlessDrive(drive: Target): drive is DriverlessDrive {
+	return (drive as DriverlessDrive).link !== undefined;
+}
+
+function isDrivelistDrive(
+	drive: Target,
+): drive is scanner.adapters.DrivelistDrive {
+	return typeof (drive as scanner.adapters.DrivelistDrive).size === 'number';
+}
+
+const ScrollableFlex = styled(Flex)`
+	overflow: auto;
+
+	::-webkit-scrollbar {
+		display: none;
+	}
+
+	> div > div {
+		/* This is required for the sticky table header in TargetsTable */
+		overflow-x: visible;
+	}
+`;
+
+const TargetsTable = styled(({ refFn, ...props }) => {
+	return (
+		<div>
+			<Table<Target> ref={refFn} {...props} />
+		</div>
+	);
+})`
+	[data-display='table-head'] [data-display='table-cell'] {
+		position: sticky;
+		top: 0;
+		background-color: rgb(248, 249, 253);
+		padding: 6px 8px;
+		color: #2a506f;
+	}
+
+	[data-display='table-cell']:first-child {
+		padding-left: 15px;
+	}
+
+	[data-display='table-body']
+		> [data-display='table-row']
+		> [data-display='table-cell'] {
+		padding: 6px 8px;
+		color: #2a506f;
+	}
```

Comment:
This whole style can be refactored to:
```css
[data-display='table-head'] [data-display='table-cell'] {
  position: sticky;
  top: 0;
}

[data-display='table-cell']:first-child {
  padding-left: 15px;
}

&& [data-display='table-row']
  > [data-display='table-cell'] {
  padding: 6px 8px;
  color: #2a506f;
}
```
so the last rule is valid for both table head and body and is more specific than rendition's rules (which override the padding in the `head`)

Reply:
:+1: missing `background-color: rgb(248, 249, 253);` for the header